File: tools/conver/tests/workspacify-reverse/fixtures/languages/javascript/src/widget.js
/**
 * A prototype-based type, and the two constructs that make it hard to read.
 *
 * `render` is attached to the prototype rather than declared in a class body, so
 * the methods of a `Widget` are not enumerated by the constructor. The registry
 * below is keyed by a computed property name, so the key is a value evaluated at
 * run time rather than a literal a reader can resolve from the syntax alone.
 */

/** The property the registry is keyed by. Its value is only known once this runs. */
const widgetKey = 'widgets';

/** Every widget ever constructed, keyed by a computed name. */
const registry = {
  [widgetKey]: [],
};

/** A widget carrying an identifier and a rendered label. */
// [::TICKET::] P24-1 changes. Details: `node .claude/scripts/tickets/show-ticket-context.js --ticket-key=P24-1 --for-spec --no-implementation-order`.
function Widget(id) {
  this.id = id;
  this.label = '';
}

Widget.prototype.render = function render() {
  this.label = `widget-${this.id}`;
  registry[widgetKey].push(this);
  return this.label;
};

/** Every widget that has been rendered, in the order it was rendered. */
// [::TICKET::] P24-1 changes. Details: `node .claude/scripts/tickets/show-ticket-context.js --ticket-key=P24-1 --for-spec --no-implementation-order`.
function renderedWidgets() {
  return registry[widgetKey];
}

module.exports = { Widget, renderedWidgets };

File: tools/conver/tests/workspacify-reverse/fixtures/languages/python/src/widget.py
"""The module whose attributes are only partly written here.

`__getattr__` at module level answers for names this file does not declare, so the
set of attributes a consumer can reach is larger than the set a reader can
enumerate from the module body.
"""

from registry import REGISTRY, Widget
from tracing import traced, CALLS

# Names this module answers for without declaring them.
LAZY_ATTRIBUTES = {
    "registry": lambda: REGISTRY,
    "call_log": lambda: list(CALLS),
}


@traced
# [::TICKET::] P24-1 changes. Details: `node .claude/scripts/tickets/show-ticket-context.js --ticket-key=P24-1 --for-spec --no-implementation-order`.
def render(identity):
    """Build a widget and return the label it renders."""
    return Widget(identity).label()


# [::TICKET::] P24-1 changes. Details: `node .claude/scripts/tickets/show-ticket-context.js --ticket-key=P24-1 --for-spec --no-implementation-order`.
def __getattr__(name):
    """Answer for a name the module body does not declare."""
    if name in LAZY_ATTRIBUTES:
        return LAZY_ATTRIBUTES[name]()
    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
